### PR TITLE
Large-scale refactor

### PR DESCRIPTION
**Pull request by nstarman**
Getting this to look like Gala.
What do you say we loop Adrian in on this and coordinate a shared repo. With not too much more work this can be nearly a gala clone.

Sorry for the stuff that says my name on it. I used https://github.com/scientific-python/cookie to set up a clean new repo and then copied the git history into it. We should fix the attribution, etc.

I also deleted all the notebooks. Might want to copy-paste them elsewhere before merging!

Also, this is a WIP. I used equinox to make this stateful, but I haven't finished all the stuff.
In particular, I haven't separated out the stream generation stuff into a separate class, like in gala.

### CODE REVIEW

File: src/galdynamix/potential/_potential/composite.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import jax.numpy as xp
+import jax.typing as jt
+
+from galdynamix.utils import jit_method
+
+from .base import PotentialBase
+
+
+class CompositePotential(PotentialBase):
```
Comment (nstarman) [inline src/galdynamix/potential/_potential/composite.py:16]: We can get this to be a dict, I'm sure of it! I just haven't tried yet.

Comment (nstarman) [reply src/galdynamix/potential/_potential/composite.py]: Done.

**Comment by adrn** [comment]
I haven't looked at the code here, but I love the idea! I started a similar project (gala-jax) to collect some ideas I've been thinking about and just invited you. Would be great to collaborate over there on an evolution of gala!

**Comment by nstarman** [comment]
If we're going to move this to gala-jax, then I'll close this PR and use my fork as a separate playground.

**Pull request closed without merging**